mods/default/workbench.lua

```diff
@@ -0,0 +1,234 @@
+	local msg = "Rename Item"
+	local text = minetest.get_meta(pos):get_string("text")
+
+	if error then
+		msg = minetest.colorize("red", error)
+	end
+
+	return
+		"size[8,7]" ..
+		default.gui_bg ..
+		default.gui_bg_img ..
+		default.gui_slots ..
+		"field[0.5,0.5;6.2,1;text;"..msg..";"..minetest.formspec_escape(text).."]" ..
+		"button[6.5,0.2;1.5,1;rename;Rename]" ..
+		"list[context;input;1.5,1.4;1,1;]" ..
+		"image[2.5,1.4;1,1;gui_workbench_plus.png]" ..
+		"image[3.5,1.4;1,1;default_nametag_slot.png]" ..
+		"list[context;nametag;3.5,1.4;1,1;]" ..
+		"image[4.5,1.4;1,1;gui_furnace_arrow_bg.png^[transformR270]" ..
+		"list[context;output;5.5,1.4;1,1]" ..
+		"list[current_player;main;0,2.85;8,1;]" ..
+		"list[current_player;main;0,4.08;8,3;8]" ..
+		"field_close_on_enter[text;false]" ..
+		default.get_hotbar_bg(0,2.85)
+end
+
+local function get_item_desc(stack)
+	if not stack:is_known() then
+		return
+	end
+
+	local desc = stack:get_meta():get_string("description")
+
+	if desc == "" then
+		desc = minetest.registered_items[stack:get_name()].description
+	end
+
+	return desc
+end
+
+local function workbench_update_text(pos, stack)
+	local meta = minetest.get_meta(pos)
+	meta:set_string("text", get_item_desc(stack))
+	meta:set_string("formspec", get_workbench_formspec(pos))
+end
+
+local function workbench_update_help(pos, type, string)
+	local meta = minetest.get_meta(pos)
+	meta:set_string("formspec", get_workbench_formspec(pos, string))
+	meta:set_string("error", type)
+end
+
+--
+-- Node definition
+--
+
+minetest.register_node("default:workbench", {
+	description = "Workbench",
+	tiles = {"default_workbench_top.png", "default_wood.png", "default_workbench_sides.png",
+		"default_workbench_sides.png", "default_workbench_sides.png", "default_workbench_sides.png"},
+	groups = {choppy = 3, oddly_breakable_by_hand = 2, flammable = 3},
+	sounds = default.node_sound_wood_defaults(),
+
+	drawtype = "nodebox",
+	paramtype = "light",
+	node_box = {
+		type = "fixed",
+		fixed = {
+			{-0.4375, 0.1875, -0.4375, 0.4375, 0.5, 0.4375}, -- Top
+			{-0.5, 0.25, -0.5, -0.4375, 0.5, 0.5}, -- TopSide1
+			{0.4375, 0.25, -0.5, 0.5, 0.5, 0.5}, -- TopSide2
+			{-0.5, 0.25, 0.4375, 0.5, 0.5, 0.5}, -- TopSide3
+			{-0.5, 0.25, -0.5, 0.5, 0.5, -0.4375}, -- TopSide4
+			{-0.4375, -0.5, 0.25, -0.25, 0.5, 0.4375}, -- Leg1
+			{-0.4375, -0.5, -0.4375, -0.25, 0.5, -0.25}, -- Leg2
+			{0.25, -0.5, 0.25, 0.4375, 0.5, 0.4375}, -- Leg3
+			{0.25, -0.5, -0.4375, 0.4375, 0.5, -0.25}, -- Leg4
+		}
+	},
+
+	on_construct = function(pos)
+		local meta = minetest.get_meta(pos)
+		meta:set_string("formspec", get_workbench_formspec(pos))
+		local inv = meta:get_inventory()
+		inv:set_size("input", 1)
+		inv:set_size("nametag", 1)
+		inv:set_size("output", 1)
+	end,
+	can_dig = function(pos, player)
+		local inv = minetest.get_meta(pos):get_inventory()
+
+		if inv:is_empty("input") and inv:is_empty("nametag") and inv:is_empty("output") then
+			return true
+		else
```

Comment:
`return inv:is_empty("input") and inv:is_empty("nametag") and inv:is_empty("output")`